src/server/service/roles.js

```diff
@@ -0,0 +1,33 @@
+const { Router } = require('express');
+
+const {
+    validateRequestUsernameAndEmail,
+    validateRequestBody
+} = require('../util/requestValidation');
+const {
+    getUserRoles,
+    addUserRoles,
+    deleteUserRoles
+} = require('../util/rolesHelpers');
+
+const rolesRouter = Router();
+
+rolesRouter.get('/roles', [
+    validateRequestBody,
+    validateRequestUsernameAndEmail,
+    getUserRoles
+]);
+
+rolesRouter.post('/roles/add', [
+    validateRequestBody,
+    validateRequestUsernameAndEmail,
+    addUserRoles
+]);
+
+rolesRouter.post('/roles/delete', [
```

Comment:
Nonblocking: Curious if we want to use DELETE instead of POST?

Reply:
Good thought! I'll add a task to make the http requests more specific. We could also use PUT for edit APIs.